_posts/2022-07-11-kubeflow-machine-learning-on-kubernetes.md

```diff
@@ -0,0 +1,145 @@
+- Kubeflow Pipelines: KFP is the component of Kubeflow I have personally used the most and also how I first discovered Kubeflow. KFP allows you to easily manage, create and deploy end-to-end scalable machine learning workflows. Think of this as an orchestrator for machine learning pipelines making experimentation and deployments pretty easy.
+- Katib: Allows you to work with automated machine learning tasks, hyperparameter tuning, and neural architecture search and supports various machine learning frameworks including PyTorch, TensorFlow, MXNet, and XGBoost. Hyperparameters are important parameters that cannot be directly estimated from the data or could be learned from the data. Finding the right values for hyperparameters is much dependent on previous experience and experimentation which is made easy with Katib. A rather detailed article by me about this can be found at [^Dagli2021]. Neural Architecture Search allows you to automate the process of designing a neural network [^Elsken2019][^Stanley2002]. Katib supports a ton of AutoML algorithms to help you do the search efficiently including but not limited to Hyperband [^Li2017], Random Search, Differentiable architecture search [^Liu2018] and more.
+- Training Operators: These allow you to train your machine learning models in a distributed fashion in Kubeflow. At the moment, training operators support TensorFlow, PyTorch, MXNet, and XGBoost jobs which are essentially Kubernetes custom resources, a way to extend the Kubernetes API. The training operator also contains the Message Passing Interface Operator to make it particularly easy to work with all reduce style distributed training jobs, an AllReduce operation synchronizes neural network parameters between separate training processes after each step of the optimization, it does so by aggregating tensors across all the devices, adding them up, and then makes them available on each device. Though you can also use the `tf.distribute` or the `torch.distributed` APIs, these are of course framework-specific [^Gibiansky2017][^Sergeev2018]. Finally, the training operator also includes supporting a job to run multiple pods with the [Volcano scheduler](https://github.com/volcano-sh/volcano).
+- Multi Tenancy: Kubeflow is built in a manner to allow multiple people to easily collaborate on a project. Multi tenancy is built around user namespaces and user access is defined through RBAC (role-based access control) policies which already exist in Kubernetes. You could easily restrict each user to only worry about and interact with the Kubeflow components in their configuration.
```

Comment:
Would be great if this part is spaced like this -> 
![image](https://user-images.githubusercontent.com/25551553/179789116-df37c60b-211c-49e6-897f-87b60aecf3c3.png)

and side headings like "Training Operators:" be mentioned using "**Training Operators**" will help with better redability.